ansible/bootstrappers.yml

```diff
@@ -0,0 +1,16 @@
+---
+- hosts: bootstrappers
+  become: yes
+  remote_user: "{{ deploy_user }}"
+  roles:
+    - { role: ssh-keys,
+          ssh_user: "{{ deploy_user }}" }
+    - { role: geerlingguy.docker,
+          docker_install_compose: false }
+    - role: python-dependencies
+    - { role: nginx_conf_bootstrappers }
+    - { role: node_exporter,
+          version: "v0.18.1" }
+    - role: go-ipfs-deploy-docker
+  vars_files:
+    - ../secrets_secure/vault.yml
```

Comment:
> Thanks, I've now had some time to go through.
> 
> Additional notes:
> 
>     * I'd like to move this to gateway/infra
This code is not specific to gateways only. Some of the roles are for setting up bootsrappers and preloaders. 

>     * Should decide on a deployment approach. If it's docker, no need to keep additional code around (and vice-versa). I'm leaning towards without (better perf, more simplicity, nginx and node_exporter available off the shelf).

I disagree that Docker will introduce any performance hit, please provide some metrics to back this up. Also Docker actually simplifies the deploy process by removing the need to install/update dependencies on the hosts themselves. I've already mentioned the metrics we need to collect from the individual go-ipfs instances, which Docker also buys us.
  
>     * It might also be useful to have Go installed in those machines so that they can be used 
>     * We should not assume that terraform gives a pre-configured machines (in terms of users, ipfs config etc). (we want to remove changing identities). We should be able to run ansible on a clean machine and have an ipfs gateway working.

Agreed, this is a first step towards the goal of moving all configuration from user-data (Terrafrom) to Ansible. For now, this is only meant to support redeploying go-ipfs to preloaders, bootstrappers, gateways as well as setup some one-off things like SSL certs on the gateways and node-exporter on the bootstrappers.  It is not meant to be a complete replacement of all the existing deployment tools but I decided to PR it so that I don't end up working on a long-running branch for months.